Add plugin: FolderFile Splitter

XuQuan-nikkkki opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

XuQuan-nikkkki commented:
For the optional feedback, I have updated the code to use `app.fileManager.trashFile(file: TAbstractFile)` as suggested in https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/commit/bfd0416ddfc30721ecf4e02ad2878c173c319626

For the required feedback regarding type checking, the `isFile` and `isFolder` method I used is a custom utility method that internally checks instanceof TFile and TFolder (https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/main/src/utils.ts#L4). Since it achieves the same functionality and improves readability in my implementation, I believe this should not be an issue.

/skip The required check on type casting, as isFile is a wrapper around instanceof TFile and ensures proper type checking. Let me know if any further clarification is needed.

joethei commented:
I don't see your `isFile` helper being called in the context where the bot is flagging the `as` usage.

XuQuan-nikkkki commented:
Thank you for your feedback. I believe the confusion may arise from the context in which isFile is being used. While it might not be directly visible in the specific lines you are referring to, the isFile (and similarly isFolder) methods are being used in other parts of the plugin where files and folders are being checked.

For example, here’s the relevant code snippet where `isFile` and `isFolder` is called in https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/df100c7e2d0cf1765ba7f691e6d1ace6e2739697/src/store.ts#L98: 
![image](https://github.com/user-attachments/assets/cd1f046d-483b-4f28-88a4-cbefb193bdff) and in https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/df100c7e2d0cf1765ba7f691e6d1ace6e2739697/src/store.ts#L109
![image](https://github.com/user-attachments/assets/03e6c6f4-4c31-4931-b4ad-18752e85122e)


I hope this clears up the misunderstanding, and I’m happy to provide any further clarification if needed.

XuQuan-nikkkki commented:
Thank you for your feedback. I believe the confusion may arise from the context in which isFile is being used. While it might not be directly visible in the specific lines you are referring to, the isFile (and similarly isFolder) methods are being used in other parts of the plugin where files and folders are being checked.

For example, here’s the relevant code snippet where `isFile` and `isFolder` is called in https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/df100c7e2d0cf1765ba7f691e6d1ace6e2739697/src/store.ts#L98: 
![image](https://github.com/user-attachments/assets/cd1f046d-483b-4f28-88a4-cbefb193bdff) and in https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/df100c7e2d0cf1765ba7f691e6d1ace6e2739697/src/store.ts#L109
![image](https://github.com/user-attachments/assets/03e6c6f4-4c31-4931-b4ad-18752e85122e)


I hope this clears up the misunderstanding, and I’m happy to provide any further clarification if needed.

/skip Clarify usage of isFile method and provide additional context for review

joethei commented:
What the bot flagged is still valid, the code it linked too and the code you linked to are in two separate places.

["name": "Apple Style Notes",](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/manifest.json#L3)
Please don't mention other product / brand names in the name if it's only a lookalike.

[name: "Open File Tree View",](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/src/main.ts#L30), [.setName("Expand Folder on Click")](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/src/SettingTab.tsx#L25), [.setName("Include Subfolder Files Count")](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/src/SettingTab.tsx#L40)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[this.detachFileTreeLeafs();](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/src/main.ts#L74), [detachFileTreeLeafs = () => {](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/src/main.ts#L104-L110)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>

[findFolderByPath: (path: string): TFolder | undefined => {](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/src/store.ts#L95-L97)
Existing `getFolderByPath` function from the Obsidian API.

[export const saveSettingsToLocalStorage = (](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/src/utils.ts#L32-L54)
Please use `loadData()` and `saveData()` from the Obsidian API for storing your data.
This allows the data to be compatible with syncing platforms like Obsidian Sync, an example can be found in the sample plugin.
Storing settings in local storage has side effects, mainly that they get synced between all vaults on the local device.

[const abstractFile = plugin.app.vault.getAbstractFileByPath(](https://github.com/XuQuan-nikkkki/apple-style-notes-plugin/blob/4f8ad1f05a4520c98644c8ea978c48e7431e7ae0/src/store.ts#L280-L282)
This part of the code is quite pointless, you already pass the `TFile` as a parameter, just use that.

XuQuan-nikkkki commented:
Fixed: instanceof type-checking issue flagged by the bot
- https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/commit/b49077219345cf829bff05fd5317989094fe5aaf

Fixed: Removed brand name from the plugin name
- https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/commit/8a88631cce5f496caa807bf801db23acbd4edf02

Fixed: Updated UI text to sentence case
- https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/commit/8520ef40b0c837141a03f5a5c871454daafdd221

Fixed: Removed detachFileTreeLeafs call from onunload
- https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/commit/f4d732fa0e69ad926f03e9b98e9101dc686e02ee

Fixed: Replaced findFolderByPath with Obsidian's getFolderByPath API
- https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/commit/86273676ad64cf191996353150df2ac2158b9beb

Fixed: Replaced localStorage with loadData() and saveData() for better sync compatibility
- https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/commit/a52decd84ab7ac23969bfcb91ef09fce41bafcb0

Fixed: Used file parameter directly instead of retrieving it via getAbstractFileByPath
- https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/commit/482e94f9535ac963736826106c0c2bbc5bd73e42

joethei commented:
["minAppVersion": "0.15.0",](https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/blob/447b360dd6d9484ffc538941b17433386c870940/manifest.json#L5)
You are using newer API's, set this to at least `1.5.7`

XuQuan-nikkkki commented:
> ["minAppVersion": "0.15.0",](https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/blob/447b360dd6d9484ffc538941b17433386c870940/manifest.json#L5) You are using newer API's, set this to at least `1.5.7`

fixed in https://github.com/XuQuan-nikkkki/FolderFile-Splitter-Plugin/commit/2277b5c3799392f91e73d852cd395566a30e4e95
